pkg/controller/template/render.go

```diff
@@ -79,22 +79,17 @@ func generateMachineConfigs(config *renderConfig, templateDir string) ([]*mcfgv1
 	return cfgs, nil
 }
 
+// generateMachineConfigsForRole is part of generateMachineConfigs; it operates
+// on a specific role which has a set of builtin templates.
 func generateMachineConfigsForRole(config *renderConfig, role string, path string) ([]*mcfgv1.MachineConfig, error) {
+	// Start with the cluster-derived configs
+	cfgs := generateDerivedMachineConfigs(config, role)
```

Comment:
Thinking about this, I think the func call (to your new function) should be after the error check on the ReadDir(path) since we're picking up the roles by walking through the dirs. So I think the ReadDir should execute successfully until we add our config to cfgs.